app/bundles/EmailBundle/Entity/EmailRepository.php

```diff
@@ -407,10 +407,19 @@ public function getEmailList($search = '', $limit = 10, $start = 0, $viewOther =
         }
 
         if ($topLevel) {
-            if (true === $topLevel || 'variant' == $topLevel) {
-                $q->andWhere($q->expr()->isNull('e.variantParent'));
-            } elseif ('translation' == $topLevel) {
-                $q->andWhere($q->expr()->isNull('e.translationParent'));
+            // BC layer
+            if (true === $topLevel || '1' === $topLevel) {
```

Comment:
todo: We should definitely refactor it. If a variable like `$topLevel` is of type `boolean`, `string`, or `numeric`, it’s a bad practice, and it indicates a space for refactoring.

Reply:
This is already the refactored version.

The condition `if (true === $topLevel || '1' === $topLevel) {` is an **intentional BC layer** to support existing and custom implementations where `$topLevel` was passed as a boolean or string. After this check, the value is normalized to a single, predictable structure and the rest of the logic operates on that.

This can be removed in a future major version once non-array usage is deprecated.